usr.bin/paste/paste.c

```diff
@@ -98,51 +114,50 @@ main(int argc, char *argv[])
 		delim = tab;
 	}
 
-	if (seq)
-		rval = sequential(argv);
-	else
-		rval = parallel(argv);
+	for (filecnt = 0; *argv; ++argv, ++filecnt) {
+		failed = 0;
+		if ((lp = malloc(sizeof(LIST))) == NULL)
+			err(EXIT_FAILURE, NULL);
+		if ((*argv)[0] == '-' && !(*argv)[1])
+			lp->fp = stdin;
+		else if (!(lp->fp = fopen(*argv, "r"))) {
+			if (!seq)
+				err(EXIT_FAILURE, "%s", *argv);
+			else
+				failed = errno;
+		}
+
+		if (cap_rights_limit(fileno(lp->fp),
```

Comment:
Maybe caph_rights_limit?

Reply:
Absolutely, but for the same reasons mentioned earlier, I felt it might be better to stick with the raw API. I’ve reviewed the helpers you’ve written, and they’re indeed excellent—adding no overhead or unnecessary abstraction. However, I thought it might still be worthwhile to use the raw API directly here. If you believe otherwise, I’m happy to adapt based on your recommendation.

Reply:
There is no need to use a raw API, capsicum helpers make it simpler.
Ideally when we make capsicum mandatory in FreeBSD it will also make code cleanup much simpler.

Reply:
Makes sense. OK, I can take care of it as well, rog.